### ISSUE TITLE
Segfault when attempting to install a package on hold

### ISSUE THREAD

**Issue by lemmi** [voidlinux/xbps#143]
My packages on hold:

    # xbps-query -H
    linux-4.2_1
    linux-headers-4.2_1
    xf86-input-evdev-2.10.0_2
    xorg-server-1.17.4_2

xbps debug output

    # xbps-install -d xf86-input-evdev

https://gist.github.com/anonymous/5a91a754ed17a55137ff

backtrace

    # gdb --args xbps-install xf86-input-evdev

https://gist.github.com/anonymous/3abfcfb1215aa730342a

**Comment by yupi2** [comment]
Pinging @xtraeme for suggestions.

It's crashing because the [`reason`](https://github.com/voidlinux/xbps/blob/b3c7d27c6b9ffddda9914a9110680df91c6b17ff/lib/repo_pkgdeps.c#L131) char* is not being set in the branch [here](https://github.com/voidlinux/xbps/blob/b3c7d27c6b9ffddda9914a9110680df91c6b17ff/lib/repo_pkgdeps.c#L238). It could also crash if the execution passes through [here](https://github.com/voidlinux/xbps/blob/b3c7d27c6b9ffddda9914a9110680df91c6b17ff/lib/repo_pkgdeps.c#L246). `reason` is eventually passed through a couple layers of functions and `strlen()` and the program crashes because `reason` is a NULL-pointer.

I wrote an [incomplete commit](https://github.com/yupi2/xbps/commit/662ba3602fd60edbed9c2004b9f62fd5738b5606) that would return `ENODEV` for [here](https://github.com/voidlinux/xbps/blob/24af8cde9b9de7b0e930e46e91b681f425792f37/bin/xbps-install/transaction.c#L307) but it seems like `reason = "hold";` could be done.

What should be done for this @xtraeme?

**Comment by Gottox** [comment]
Unfortunately I can't reproduce your error. Can you edit the [test case](https://github.com/voidlinux/xbps/blob/master/tests/xbps/xbps-install/behaviour_tests.sh) to trigger this error?

**Comment by yupi2** [comment]
@Gottox
[This](https://github.com/voidlinux/xbps/pull/144) is working for me.
The test should be renamed to something else like 'update_pkg_with_held_dep'.
Below shows what triggered the crash:

`xf86-input-evdev-2.10.0_3` requires `xserver-abi-input-22_1`.
`xorg-server-1.18.0` (revision 1-6 so far) provides `xserver-abi-input-22_1`.
`xorg-server` is held at `xorg-server-1.17.4_2` with `xserver-abi-input-21_1` provided.

TLDR: **pkgA-1.1_1** requires **pkgB-22_1**. **pkgC** provides **pkgB-21_1**. **pkgC** is held on an older version so **pkgB** can't be updated from version *21_1*.

**Comment by xtraeme** [comment]
Working on this right now, fix incoming.
